index.js

```diff
@@ -1,6 +1,25 @@
 /* jshint node: true */
 'use strict';
 
+var path = require('path');
+var mergeTrees = require('broccoli-merge-trees');
+
 module.exports = {
-  name: 'lodash'
+  name: 'lodash',
+
+  treeForAddon: function(tree) {
+    var lodashPath = path.join(this.project.root, 'node_modules', 'ember-lodash', 'node_modules', 'lodash-es');
+
+    if (this.app.name === 'dummy') {
+      lodashPath = path.join(this.project.root, 'node_modules', 'lodash-es');
+    }
+
+    var lodashTree = this.treeGenerator(lodashPath);
+
+    var trees = mergeTrees([lodashTree, tree], {
```

Comment:
addon tree wins out over the lodash-es tree, so you're able to overwrite the bits that do not work, like `addon/internal/root.js`